url913d8566d1_AnsBalin_rbn_tgz-pJ8-MainJ8/src/DataOutput.java

```diff
@@ -0,0 +1,98 @@
+import java.io.*;
+import java.util.*;
+
+public class DataOutput{
+
+  private String path;
+  private boolean append = false;
+
+  public DataOutput(String path){
+    
+    this.path = path;
+    
+  }
+  
+  public DataOutput(String path, boolean append){
+  
+    this.path = path;
+    this.append = append;
+    
+  }
+  
+  public String getPath(){
+    
+    return path;
+    
+  }
+  
+  public void writeToFile(int[] arr) throws IOException{
+  
+    FileWriter write = new FileWriter(path, append);
+    PrintWriter print_line = new PrintWriter(write);
+    
+    for(int i=0; i<arr.length; i++){
+      print_line.printf("%d\t", arr[i]);
+    }
+    print_line.printf("\n");
+
+    print_line.close();
+  
+  }
+  
+  public void writeToFile(double[] arr) throws IOException{
+    
+    FileWriter write = new FileWriter(path, append);
+    PrintWriter print_line = new PrintWriter(write);
+    
+    for(int i=0; i<arr.length; i++){
+      print_line.printf("%f\t", arr[i]);
+    }
+    print_line.printf("\n");
+    
+    print_line.close();
+    
+  }
+  
+  public void writeToFile(String str) throws IOException{
+  
+    FileWriter write = new FileWriter(path, append);
+    PrintWriter print_line = new PrintWriter(write);
+    
+    print_line.printf("%s\n", str);
+    print_line.close();
+  
+  }
+  
+  public void writeToFile(Object obj) throws IOException{
+  
+    FileOutputStream fos = new FileOutputStream(path, append);
+    ObjectOutputStream oos = new ObjectOutputStream(fos);
+    
+    oos.writeObject(obj);
+    
+    oos.flush();
+    oos.close();
+  
+  }
+  
+  public Network readFile() throws IOException, ClassNotFoundException{
+    
+    FileInputStream fis = new FileInputStream(path);
+    ObjectInputStream ois = new ObjectInputStream(fis);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *ois*. Execution paths that do not contain closure statements were detected. Ensure *ois* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).